### PR TITLE
Storybook: Created apps-list-component

### PR DESCRIPTION
**Pull request by Douglashdaniel**
Resolves <https://github.com/brave/brave-browser/issues/14955>

Created an apps list component and added it too storybook.
Also created a reusable search bar component that was needed for the apps panel.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. `npm run storybook`
2. Confirm that the storybook shows new component.
3. Check functionality of the new component.

https://user-images.githubusercontent.com/40611140/113009437-2d0c3100-9135-11eb-8fbc-46d70c545318.mov

### CODE REVIEW

**Review by jamesmudgett — APPROVED** [review]
UI looks great

File: components/brave_wallet_ui/mock-data/apps-list.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    appList: [
+      {
+        name: 'Compound',
+        description: 'Unlock a universe of open financial applications.',
+        url: 'https://app.compound.finance/',
+        icon: compoundIcon
+      },
+      {
+        name: 'MakerDAO',
+        description: 'Maker - stablecoin, loans and governance...',
```
Comment (bbondy) [inline components/brave_wallet_ui/mock-data/apps-list.ts:20]: I think all these Descriptions should also be marked as localizable

File: components/brave_wallet_ui/stories/wallet-extension-panels.tsx [hunk 2]
```diff
@@ -118,6 +126,44 @@ export const _ConnectedPanel = () => {
...
+      const filteredList = mergedList.filter((app) => {
+        return (
+          app.name.toLowerCase() === search.toLowerCase() ||
+          app.name.toLowerCase().startsWith(search.toLowerCase()) ||
+          app.name.toLowerCase().includes(search.toLowerCase())
+        )
+      })
+      const newList = [
+        {
+          category: 'Search Results',
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/stories/wallet-extension-panels.tsx]: Done! I added locale to all pertaining components in this PR and will create a new issue/pr to add locale to the rest! :)

File: components/brave_wallet_ui/mock-data/apps-list.ts (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/mock-data/apps-list.ts]: Added! :)

**Review by bbondy — APPROVED** [review]

**Pull request merged**